src/tests/presets/test_erc20votes.cairo

```diff
@@ -0,0 +1,451 @@
+use integer::BoundedInt;
+use openzeppelin::tests::extensions::test_erc20votes::{
+    assert_event_delegate_changed, assert_only_event_delegate_changed,
+    assert_event_delegate_votes_changed, assert_only_event_delegate_votes_changed
+};
+use openzeppelin::tests::utils::constants::{
+    NAME, SYMBOL, DECIMALS, SUPPLY, VALUE, ZERO, OWNER, SPENDER, RECIPIENT
+};
+use openzeppelin::tests::utils;
+use openzeppelin::token::erc20::presets::ERC20VotesPreset;
+use openzeppelin::token::erc20::presets::ERC20VotesPreset::ERC20Impl;
+use openzeppelin::token::erc20::presets::ERC20VotesPreset::VotesImpl;
+use openzeppelin::utils::structs::checkpoints::Checkpoint;
+use starknet::ContractAddress;
+use starknet::contract_address_const;
+use starknet::testing;
+use traits::Into;
+use zeroable::Zeroable;
+
+//
+// Setup
+//
+
+fn STATE() -> ERC20VotesPreset::ContractState {
+    ERC20VotesPreset::contract_state_for_testing()
+}
+
+fn setup() -> ERC20VotesPreset::ContractState {
+    let mut state = STATE();
+    testing::set_block_timestamp('ts0');
+    ERC20VotesPreset::constructor(ref state, NAME, SYMBOL, SUPPLY, OWNER());
+    utils::drop_events(ZERO(), 2);
+    state
+}
+
+//
+// constructor
+//
+
+#[test]
+#[available_gas(2000000)]
+fn test_constructor() {
+    let mut state = STATE();
+    ERC20VotesPreset::constructor(ref state, NAME, SYMBOL, SUPPLY, OWNER());
+
+    assert(ERC20Impl::balance_of(@state, OWNER()) == SUPPLY, 'Should eq inital_supply');
+    assert(ERC20Impl::total_supply(@state) == SUPPLY, 'Should eq inital_supply');
+    assert(ERC20Impl::name(@state) == NAME, 'Name should be NAME');
+    assert(ERC20Impl::symbol(@state) == SYMBOL, 'Symbol should be SYMBOL');
+    assert(ERC20Impl::decimals(@state) == DECIMALS, 'Decimals should be 18');
+}
+
+//
+// Getters
+//
+
+#[test]
+#[available_gas(2000000)]
+fn test_total_supply() {
+    let mut state = setup();
+    assert(ERC20Impl::total_supply(@state) == SUPPLY, 'Should eq SUPPLY');
```

Comment:
In the meantime, I think we should be able to import a function from one test and call it in another, this may help us to avoid re-writing some tests, by designing some simple pattern around this.